suite-native/module-settings/src/utils.ts

```diff
@@ -0,0 +1,7 @@
+import { SelectItemType } from '@suite-native/atoms';
+import { Currency } from '@suite-common/suite-config';
+
+export const transformFiatCurrencyToSelectItem = (fiatCurrency: Currency): SelectItemType => ({
```

Comment:
I don't think it's worth having this separated in utils file. This function is super simple and  `CurrencySelector` is very small component co it could be located directly in component. There is also very small chance that it will be used anywhere else that in this component, so another reason why keep it together.

Utils file should be used primary for utils that are reused across multiple files or if utils is very long and complicated. This is general rule applied to anything.

Reply:
No, but I think it's nice to not have functions like this in component. The component might grow and there will also be more functions in this utils file. It's nicer IMO to have these separated because they're not part of component definition.

Reply:
1. It's usable only in that component, there is no reason to believe it will be used anywhere else, if so it could be moved easily later
2. It's part of component functionality - it preparing very specific data shape for specific this very specific component. It's worth to break context to save 5 lines of code and force people to go to different file when they want to understand what is happening?
3. This utils files for everything that could be function leads to something I call "garbage files". File where you can throw anything and call it utils. In the ends you have very long file with lot unrelated functions that are out of context it's usage and you need to search it. That's much worse than little longer component file.
4. This garbage files for everything produces dead code. I seen this too many times, people remove or refactor component so they don't need that util function anymore and they only remove import. Then you have dead code which is unused.

I am really not agains using these utils files, they could be very useful, but you need to do that with some rules and rule "throw every function that could be standalone here" is definitely bad rule. I've seen so many times times files went into total mess and grow to hundreds of lines, which is crazy. And also dead code.

Reply:
https://github.com/trezor/trezor-suite/pull/5895/commits/285dce9e0a56ad98fd9ab9555c697cffc3a30539